### ISSUE TITLE
Order of Find All / Replace All buttons

### ISSUE THREAD

**Issue by nicolasnoakes** [notepad-plus-plus/notepad-plus-plus#8213]
### Position/order of Find All and Replace All buttons is likely to cause unintended mistakes

The logical sequence of the operational buttons on the "Find" and "Replace" tabs is not consistent, which will lead to users making unintended replacements.

### Steps to Reproduce the Issue

In the Find dialogue, the buttons on the "Find" tab are sequenced as follows, top to bottom:

1. Find Next
2. Count
3. **Find All in All Opened Documents**
4. **Find All in Current Document**
5. Close

But on the "Replace" tab, the logical order of the last operation buttons (buttons 3 & 4) is reversed.

Find 

1. Next
2. Replace
3. **Replace All** [in Current Document]
4. **Replace All in All Opened Documents**
5. Close

### Expected Behavior
I propose that it would be better and more logical (increasing scope from top to bottom) to swop buttons 3 & 4 on the "Find" tab. Thus the "in Current Document" and the "in All Opened Documents" buttons would be consistently positioned.

![notepad-find](https://user-images.githubusercontent.com/14802452/81004162-fc832080-8e19-11ea-8047-992729b8f355.png)

![notepad-replace](https://user-images.githubusercontent.com/14802452/81004186-06a51f00-8e1a-11ea-9398-bfb21b11ec48.png)

**Comment by Uhf7** [comment]
Yes, I'am with you. And I would even go a step further and make it look like this:
![FindDlg001](https://user-images.githubusercontent.com/40629737/81048010-d3b36780-8ebb-11ea-9faf-571011c5995c.png)
Then the **_In Selection_** checkmark can work for both **_Count_** and **_Find All_**.
And **_Find All_** is a better match to **_Replace All_** than **_Find All in Current Document_**. 
And optically, it is easier to make a difference between the **_Find All_** button and the **_Find All in All Opened Documents_** button.

**Comment by sasumner** [comment]
>Then the In Selection checkmark can work for ..... and Find All.

Note that currently *In Selection* has no effect on *Find All in Current Document* (a.k.a. *Find All* as proposed here).  It is simply a feature under consideration; see #8133

**Comment by Uhf7** [comment]
I took the function itself as implemented in PR #8207. That is why I trust this function is possible. But I didn't look at the details of the PR right now.

**Comment by nicolasnoakes** [comment]
I appreciate the hesitation to change UI (see #8133) for users who have become familiar with the interface over a long period of time, and other concerns typical with such changes.

What happens is that I use "Find All in Current Document" a lot and then usually make manual changes, but sometimes I decide that replacing every occurrence is necessary and then (if you don't pay attention) "Replace All in All Opened Documents" is the natural button to click. Kind of muscle memory like... And undoing that in all opened documents is a hassle at best, but if you don't realise what you did (and think you only affected the current document) then you can have a very big mess on your hands!

**Comment by Uhf7** [comment]
But, if the large buttons at the **Find** page are swapped, then I assume, that many users klick the wrong one accidentally, because they look very similar. I think, most poeple decide by button position, where to klick, and do not read the text on it. This actually is an argument against simply swapping the buttons.

And I don't get yet why the proposed UI change would increase the danger of clicking the **_Replace All in All Opened Documents_** button in the **Replace** page. You describe using **_Find All in Current Document_**, which would mutate to **_Find All_**, and after this, it is natural to click **_Replace In All Documents_**? For me, the natural pairing is
- **_Find All (in Current Document)_**  <---> **_Replace All (in Current Document)_**
- **_Find All in All Documents_** <---> **_Replace All in All Documents_**

Another thought: The buttons working for all opend documents have much more impact than the buttons for the current document, they are more dangerous and this is why they should be big and the other ones small.

**Comment by sasumner** [comment]
>The buttons working for all opened documents have much more impact than the buttons for the current document, they are more dangerous and this is why they should be big and the other ones small.

Making the "dangerous" buttons large seems counterintuitive.
But as I said before, all UI debate is very subjective, often with contrary views both making sense.

**Comment by Uhf7** [comment]
Ok, I should not make the "dangerous" buttons most inviting to click.

Dangerous was the wrong word here. My actual thought was, the bigger buttons do more. And I don't find them inviting. The multi-line text on it looks like "Read me carefully before click" .

There are editors with safety questions for hard-to-revert operations. Perhaps this would solve the miss-click problem more adequately, no matter which button sizes or positions.

**Comment by sasumner** [comment]
>safety questions for hard-to-revert operations

Some references:

- Point 3 here:  #2855  The poster of this issue also has other ideas for improving the UI!
- #5253

**Comment by Uhf7** [comment]
I see.

 I like point 3 in #2855, not point 1. I have no opinion about the other two.

I also like the idea of moving the **_Replace All In All Documents_** button further away from **_Replace All_** in #5253, but actually, I would prefer the safety question, no matter where the button is. Perhaps it could come with an offer to save all modified files, then undoing all replacements is easy.  A simple Yes/No message box would do it too, of course. And it doesn't even require UI changes, except in settings, if it would be made optional.

**Comment by radiowavesurfer** [comment]
+1 for the safety question on **_Replace All In All Documents_**.

I came looking to see if there was an issue for this precisely because I just had a major "Oh shit" moment after hitting the wrong key and being told that I had just done a replace on 88 documents. (I had no idea I had that many open. Shows what I know.)

What I actually did was to type **Alt+u**, mistakenly thinking that it was the shortcut for the **_Regular Expression_** radio button. I should instead have typed **Alt+g**.

There is no positioning or amount of wording on that button that would have averted my mistake, as I didn't mean to press the button in the first place.
